### DISCUSSION TITLE
Remove build env variables

### DISCUSSION THREAD

**Discussion by predragnikolic** [coollabsio/coolify#3006] [Q&A] [answered]
Currently when adding .env variables there is a checkbox to specify if an env variable is available at build time. 

Is there a real need to a checkbox to allow specifing build time env variables?
As an end user I would expect the specified env variables to be available in build time and in runtime, with no need to check them.

I would propose to remove this build env variables in favor of having the specified env variables available in build time and in runtime.

**Comment by nuxwin** [comment, marked as answer]
@predragnikolic I do think you missunderstood the feature. When deploying via git, the nixpacks will create a Dockerfile. As reminder, environment variables for the build time are defined via ARG and willn't be available in the final container, while those defined by ENV will be available inside the final container. So, some people can need build time variables only (this is when the checkbox is checked), while other people can require env variables for the runtime.  See also : https://nixpacks.com/docs/how-it-works

**Comment by peaklabs-dev** [comment]
@predragnikolic Check the comment of @nuxwin.
